File: app/controllers/Application.scala
package controllers

import com.google.inject.Inject
import models.{Admin, Admins}
import play.api.libs.json.Json
import play.api.mvc._
import models.Admin._

class Application @Inject () (admins: Admins) extends Controller {

  def index = Action {
    val all: List[Admin] = admins.all
    Ok(Json.toJson(all))
  }

}

File: app/controllers/Application.java
package controllers;

import com.google.gson.Gson;
import com.google.inject.Inject;
import daos.UserDao;
import http.actions.OAuth2Action;
import models.User;
import play.mvc.Controller;
import play.mvc.Result;

@With(OAuth2Action.class)
public class Application extends Controller {

    @Inject
    private UserDao userDao;
    @Inject
    private Gson gson;

    public Result greet() {
        String authorization = request().getHeader("Authorization");
        String[] split = authorization.split(" ");
        String accessToken = split[1];
        User u = userDao.find(accessToken);

        return ok(gson.toJson(u));
    }
}
